kpatch-build/kpatch-elf.c

```diff
@@ -328,9 +341,29 @@ void kpatch_create_symbol_list(struct kpatch_elf *kelf)
 					sym->name);
 
 			if (is_bundleable(sym)) {
-				if (sym->sym.st_value != 0)
-					ERROR("symbol %s at offset %lu within section %s, expected 0",
+				/*
+				 * function prologue generated by gcc6 on ppc64le has
+				 * the sequence:
+				 *
+				 *	.globl my_func
+				 *	.type my_func, @function
+				 *	.quad .TOC.-my_func
+				 * my_func:
+				 *	.reloc ., R_PPC64_ENTRY ; optional
+				 *	ld r2,-8(r12)
+				 *	add r2,r2,r12
+				 * 	.localentry my_func, .-my_func
+				 *
+				 * where .localentry for the functions, will generated
+				 * st_value of 0x8 for the local functions. Skip, this
+				 * case.
+				 */
+				if (sym->sym.st_value != 0) {
+					if (!gcc6_check() ||
+					   (sym->type != STT_FUNC || sym->sym.st_value != 0x8))
+						ERROR("symbol %s at offset %lu within section %s, expected 0",
 					      sym->name, sym->sym.st_value, sym->sec->name);
+				}
```

Comment:
The localentry or the local function can be guaranteed by 
1. if the ```symbol->st_shndx != SHN_UNDEF```
2. Decode the number of instruction between functions global and local entry point. This information is only available for local functions ```sym->bind == STB_LOCAL``` or is 0.

This will also remove the need for gcc6_check(). I am running a couple of test before posting the next version.